### ISSUE TITLE
The AP of test images

### ISSUE THREAD

**Issue by yoyoleeisstrong** [RookieHong/WSDDN#2]
![image](https://user-images.githubusercontent.com/8444859/104083437-7442fe00-5279-11eb-84c0-bca58a90eff4.png)

Thanks for your work to realize WSDDN using Pytorch. However, inference APs are very low in my experiment. How about your results?

**Comment by RookieHong** [comment]
Sorry, this repository was a long time ago, I don't remember the results. But I remember that I could not reproduce the results in the paper. I guess it might be because my evaluation codes are wrong. I suggest you visualize the detections to confirm whether it is the evaluation problem or the model training failure. Also, I suggest that you use the existing object detection framework to evaluate the results of the detection, such as detectron2 and mmdetection, which will get an accurate evaluation.

**Comment by yoyoleeisstrong** [comment]
I think I found out where the problem is. I used your proposals in the data file while using VOC2007 data downloaded from the internet. It seems the proposals don't fit the images. I'm gonna create proposals by myself. Btw, why do you think your evaluation code is wrong? Do the results you produce are much lower than those in the paper?

**Comment by RookieHong** [comment]
1. The proposals were generated using a python implementation from [here](https://github.com/AlpacaDB/selectivesearch), I didn't find them wrong at the time. 
2. AP is calculated under multiple IoU thresholds, while I only set one threshold here. In fact, the evaluation codes of those object detection frameworks are extremely complicated, and my code is too simple, I can not guarantee that it is correct, which is why I recommend that you directly use their codes to evaluate.
3. Yes, the AP I got is very low compared to the paper, but the visualization results look reasonable, so I think the problem lies in the evaluation code.

**Comment by yoyoleeisstrong** [comment]
Thanks for your patient explanation! I will try your suggestion! I'll let you know the accuracy of my dataset once I finished the experiment.

**Comment by soongyuu** [comment]
> I think I found out where the problem is. I used your proposals in the data file while using VOC2007 data downloaded from the internet. It seems the proposals don't fit the images. I'm gonna create proposals by myself. Btw, why do you think your evaluation code is wrong? Do the results you produce are much lower than those in the paper?

I also encountered the same problem, has not solved. Can I learn your improvement method and related program code? I would be grateful！

**Comment by hpu-dxx** [comment]
> > I think I found out where the problem is. I used your proposals in the data file while using VOC2007 data downloaded from the internet. It seems the proposals don't fit the images. I'm gonna create proposals by myself. Btw, why do you think your evaluation code is wrong? Do the results you produce are much lower than those in the paper?
> 
> I also encountered the same problem, has not solved. Can I learn your improvement method and related program code? I would be grateful！
Hello, have you solved this problem? I encountered a fatal problem when running this code: in the test code, one of the scores obtained through WSDDN was so large that after normalization, the other scores were so small.Looking forward to your reply. Thank you！

**Comment by hpu-dxx** [comment]
> > I think I found out where the problem is. I used your proposals in the data file while using VOC2007 data downloaded from the internet. It seems the proposals don't fit the images. I'm gonna create proposals by myself. Btw, why do you think your evaluation code is wrong? Do the results you produce are much lower than those in the paper?
> 
> I also encountered the same problem, has not solved. Can I learn your improvement method and related program code? I would be grateful！
Hello, have you solved this problem? I encountered a fatal problem when running this code: in the test code, one of the scores obtained through WSDDN was so large that after normalization, the other scores were so small.Looking forward to your reply. Thank you！

**Comment by hpu-dxx** [comment]
> ![图像](https://user-images.githubusercontent.com/8444859/104083437-7442fe00-5279-11eb-84c0-bca58a90eff4.png)
> 
> 感谢您使用 Pytorch 实现 WSDDN。但是，在我的实验中，推理 AP 非常低。你的结果呢？

Hello, have you solved this problem? I encountered a fatal problem when running this code: in the test code, one of the scores obtained through WSDDN was so large that after normalization, the other scores were so small.Looking forward to your reply. Thank you！

**Comment by hpu-dxx** [comment]
> * The proposals were generated using a python implementation from [here](https://github.com/AlpacaDB/selectivesearch), I didn't find them wrong at the time.
> * AP is calculated under multiple IoU thresholds, while I only set one threshold here. In fact, the evaluation codes of those object detection frameworks are extremely complicated, and my code is too simple, I can not guarantee that it is correct, which is why I recommend that you directly use their codes to evaluate.
> * Yes, the AP I got is very low compared to the paper, but the visualization results look reasonable, so I think the problem lies in the evaluation code.

Hi, I have been running your code recently, and the AP value was very low during the Eval phase. When I looked at some of the EVAL code, I felt there was something wrong with the EVAL code. So I would like to ask you which part of your eval code you said had errors? And what's your AP score? Thank you very much!
